ui-core/src/surveyUtils.tsx

```diff
@@ -74,31 +262,31 @@ type UseSurveyJsModelOpts = {
  * survey on completion and display a completion banner.  To continue displaying the form, use the
  * `refreshSurvey` function
  * @param pager the control object for paging the survey
+ * @param envName
+ * @param profile
+ * @param proxyProfile
  * @param opts optional configuration for the survey
  * @param opts.extraCssClasses mapping of element to CSS classes to add to that element. See
  * https://surveyjs.io/form-library/examples/survey-customcss/reactjs#content-docs for a list of available elements.
- * @param opts.extraVariables extra variables you might want to include for a specific survey type that would not
- * be useful for all surveys (e.g., {isProxyEnrollment} for the pre-enroll survey)
  */
 export function useSurveyJSModel(
   form: Survey,
   resumeData: SurveyJsResumeData | null,
   onComplete: () => void,
   pager: PageNumberControl,
+  envName: EnvironmentName,
+  profile?: Profile,
+  proxyProfile?: Profile,
```

Comment:
100% agree that the parameter list on this is way too burdensome. hooks are an interesting idea here, let me see what I can do to clean it up (but i may defer that work to a followup PR though since this branch is super vulnerable to merge conflicts)